pandas/_testing/asserters.py

```diff
@@ -921,6 +921,14 @@ def assert_series_equal(
         else:
             assert_attr_equal("dtype", left, right, obj=f"Attributes of {obj}")
 
+    if (
+        is_integer_dtype(left.dtype)
+        and is_integer_dtype(right.dtype)
+        and isinstance(left._values, type(right._values))
+        and isinstance(right._values, type(left._values))
```

Comment:
just for my understanding, why do we need these two lines?

Reply:
Sure.
 In this test case pandas/tests/series/test_constructors.py::TestSeriesConstructors::test_constructor_maskedarray:572
one assertion is failing.

```
    def test_constructor_maskedarray(self):
        data = ma.masked_all((3,), dtype=float)
        result = Series(data)
        expected = Series([np.nan, np.nan, np.nan])
        tm.assert_series_equal(result, expected)
    
        data[0] = 0.0
        data[2] = 2.0
        index = ["a", "b", "c"]
        result = Series(data, index=index)
        expected = Series([0.0, np.nan, 2.0], index=index)
        tm.assert_series_equal(result, expected)
    
        data[1] = 1.0
        result = Series(data, index=index)
        expected = Series([0.0, 1.0, 2.0], index=index)
        tm.assert_series_equal(result, expected)
    
        data = ma.masked_all((3,), dtype=int)
        result = Series(data)
        expected = Series([np.nan, np.nan, np.nan], dtype=float)
        tm.assert_series_equal(result, expected)
    
        data[0] = 0
        data[2] = 2
        index = ["a", "b", "c"]
        result = Series(data, index=index)
        expected = Series([0, np.nan, 2], index=index, dtype=float)
        tm.assert_series_equal(result, expected)
    
        data[1] = 1
        result = Series(data, index=index)
        expected = Series([0, 1, 2], index=index, dtype=int)
>       tm.assert_series_equal(result, expected)
E       AssertionError: Series are different
E       
E       Series classes are different
E       [left]:  MaskedArray
E       [right]: ndarray
```

Even though `left.dtype `and` right.dtype `are same their type are different. So here are setting check_exact to true only when the types of left and right are same.